sys-block/parted/parted-3.2.recipe

```diff
@@ -0,0 +1,144 @@
+data on hard disks and disk imaging."
+HOMEPAGE="https://www.gnu.org/software/parted/"
+COPYRIGHT="1999-2014 Free Software Foundation Inc."
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="http://ftp.gnu.org/gnu/parted/parted-$portVersion.tar.xz"
+CHECKSUM_SHA256="858b589c22297cacdf437f3baff6f04b333087521ab274f7ab677cb8c6bb78e4"
+PATCHES="parted-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 ?x86 ?x86_64"
+SECONDARY_ARCHITECTURES="!x86_gcc2 ?x86"
+
+PROVIDES="
+	parted$secondaryArchSuffix = $portVersion
+	cmd:parted$secondaryArchSuffix = $portVersion
+	cmd:partprobe$secondaryArchSuffix = $portVersion
+	lib:libparted$secondaryArchSuffix = 2.0.1 compat >= 2
+	lib:libparted_fs_resize$secondaryArchSuffix = 0.0.1 compat >= 0
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libintl$secondaryArchSuffix
+	lib:libncurses$secondaryArchSuffix
+	lib:libreadline$secondaryArchSuffix
+	lib:libuuid$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	parted${secondaryArchSuffix}_devel = $portVersion
+	devel:libparted$secondaryArchSuffix = 2.0.1 compat >= 2
+	devel:libparted_fs_resize$secondaryArchSuffix = 0.0.1 compat >= 0
+	"
+REQUIRES_devel="
+	parted$secondaryArchSuffix == $portVersion base
+	devel:libiconv$secondaryArchSuffix
+	devel:libintl$secondaryArchSuffix
+	devel:libncurses$secondaryArchSuffix
+	devel:libreadline$secondaryArchSuffix
+	devel:libuuid$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libiconv$secondaryArchSuffix
+	devel:libintl$secondaryArchSuffix
+	devel:libncurses$secondaryArchSuffix
+	devel:libreadline$secondaryArchSuffix
+	devel:libuuid$secondaryArchSuffix
+	"
+
+BUILD_PREREQUIRES="
+	cmd:aclocal
+	cmd:autoconf
+	cmd:autom4te
+	cmd:automake
+	cmd:autoreconf
+	cmd:awk
+	cmd:gawk
+	cmd:grep
+	cmd:iconv
+	cmd:libtool
+	cmd:make
+	cmd:makeinfo
+	cmd:gcc$secondaryArchSuffix
+#	cmd:gnulib_tool$secondaryArchSuffix	# not required
+	cmd:ld$secondaryArchSuffix
+	cmd:libtoolize
+#	cmd:gperf				# not required
+	cmd:pkg_config$secondaryArchSuffix
+#	cmd:rsync				# not required
+	cmd:gettext
+	"
+
+#USE_DEBUG_INFO_FOR_PARTED="no"
+#USE_DEBUG_INFO_FOR_PARTED="all" # not working due to haikuporter bug.
+#USE_DEBUG_INFO_FOR_PARTED="all_but_libparted-fs-resize"
+ USE_DEBUG_INFO_FOR_PARTED="all_with_workaround"
+
+case "$USE_DEBUG_INFO_FOR_PARTED" in
+	all_but_libparted-fs-resize)
+		LIBPARTED_FS_RESIZE_FILENAME=
+		;;
+	all|all_with_workaround)
+		LIBPARTED_FS_RESIZE_FILENAME=$libDir/libparted-fs-resize.so.0.0.1
+		;;
+esac
+
+if [ "$USE_DEBUG_INFO_FOR_PARTED" != no ]; then
+		defineDebugInfoPackage parted \
+			$binDir/parted \
+			$binDir/partprobe \
+			$libDir/libparted.so.2.0.1 \
+			$LIBPARTED_FS_RESIZE_FILENAME
+fi
+
+if [ "$USE_DEBUG_INFO_FOR_PARTED" = all_with_workaround ]; then
+# defineDebugInfoPackage does not produce correct {PROVIDES,REQUIRES}_debuginfo
+# when a target has dashes in its name. So we need to override these.
```

Comment:
We probably should fix this instead of having a workaround -- cc @DarkmatterVale, can you look into this?